### PR TITLE
fix(core): add proper UMD loader

### PR DESCRIPTION
**Pull request by straker**
Our UMD loader wasn't really a true UMD loader, so fixed it by using the template from https://github.com/umdjs/umd/blob/master/templates/returnExports.js. Due to our conditional loading of `axe.source` I had to do some additional wrapping of the UMD loader.

Note: this now prevents the `axe` object from going onto the global namespace if AMD or CommonJS are defined. This might be a breaking change, I'm not sure how to judge it based on the example puppeteer use. If so I can update it to always export, which means we close the linked issue as will not fix.

Closes issue: https://github.com/dequelabs/axe-core/issues/2052

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [ ] Follows the commit message policy, appropriate for next version
- [ ] Code is reviewed for security

### CODE REVIEW

**Comment by straker** [comment]
As much as I agree we cannot as Wepback is not the final target of the build. Webpack is used to gather all the imports, but we still have to build the final `axe._load` call from the JSON metadata files into `core/index.js`, which is then concatenated with the Webpack output. Until we stop using JSON metadata files to build things we cannot use Webpack to generate the UMD output.

**Comment by stephenmathieson** [comment]
> Until we stop using JSON metadata files to build things we cannot use Webpack to generate the UMD output.

I see... thanks for clarifying. I suppose this works for now then. 😢

File: test/integration/full/umd/umd-module-exports.js [hunk 1]
```diff
@@ -19,10 +24,4 @@ describe('UMD module.export', function() {
 	it('should ensure axe source includes axios', function() {
 		assert.isTrue(axe.source.includes(axe.imports.axios.toString()));
 	});
-
-	it('should include doT', function() {
```
Comment (stephenmathieson) [inline test/integration/full/umd/umd-module-exports.js:23]: Why was this removed?

File: doc/examples/puppeteer/axe-puppeteer.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ const main = async url => {
 		// Inject and run axe-core
 		const handle = await page.evaluateHandle(`
 			// Inject axe source code
-			${axeCore.source}
+			var axe = ${axeCore.source}
```
Comment (stephenmathieson) [inline doc/examples/puppeteer/axe-puppeteer.js:30]: Why is this necessary? I don't think I understand what's happening here.

Comment (straker) [reply doc/examples/puppeteer/axe-puppeteer.js:30]: Puppeteer loads [CommonsJS by default](https://github.com/puppeteer/puppeteer/issues/3685#issuecomment-453229607) so there is no global object after the source runs.

File: test/integration/full/umd/umd-module-exports.js (hunk 1 above)
Comment (straker) [reply test/integration/full/umd/umd-module-exports.js]: Mistake

File: doc/examples/puppeteer/axe-puppeteer.js (hunk 2 above)
Comment (stephenmathieson) [reply doc/examples/puppeteer/axe-puppeteer.js:30]: This is not true. The comment you linked to is discussing something completely different.

Within the `Page#evaluateHandle` context, there is no `module` defined. Here's a script proving this:

```js
const puppeteer = require('puppeteer')

const test = async () => {
  const browser = await puppeteer.launch()
  const page = await browser.newPage()

  const handle = await page.evaluateHandle(`
    Promise.resolve({
      module: typeof module,
      define: typeof define,
      window: typeof window
    })
  `)

  const types = await handle.jsonValue()
  console.log('typeof module: "%s"', types.module)
  console.log('typeof define: "%s"', types.define)
  console.log('typeof window: "%s"', types.window)

  await page.close()
  await browser.close()
}

test().catch(err => {
  console.error(err)
  process.exit(1)
})
```

This will output:

```
typeof module: "undefined"
typeof define: "undefined"
typeof window: "object"
```

I do not understand why we need to add `var axe = ` in front of `axe.source`, and believe it's due to a bug.

**Comment by straker** [comment]
This is a breaking change since axe is removed from the global scope when using AMD or CommonJS. We can't do this now and will have to wait for 5.0 to do this if we still want to.
